### DISCUSSION TITLE
iCloud China region support and 'resOriginalAltRes' syncing support.

### DISCUSSION THREAD

**Discussion by leafnsand** [steilerDev/icloud-photos-sync#279] [Ideas]
1. iCloud in China use different auth endpoint. Pls make a option for Chinese users.
2. I usually shoot in jpeg+raw format and then import files to my iPhone. iCloud treat jpeg as 'resOriginalRes', raw as 'resOriginalAltRes'. Pls sync this version of photo as well. And maybe you can also make a option for people who want to sync builtin-photo-app-modified version jpeg, which is saved as 'resJPEGFull' (?just my guess).

**Comment by steilerDev** [comment]
@leafnsand do you have any insights on the endpoints used by iCloud in China? Happy to look into adding this down the road, but need some help on this, since I don't have access to a Chinese iCloud account.

On the various options of assets: I need to check how much work this is going to be - currently I got 'original' and 'edited' - I'd need to think about how to handle this additional complexity

**Comment by leafnsand** [comment]
@steilerDev I simply replaced all URL end with `[apple | icloud].com` with `[apple | icloud].com.cn` and started sync. Then I got this error below. As I know, the only different between iCloud global and iCloud China is the endpoint URL. And some tools like [icloudpy](https://github.com/mandarons/icloudpy) and `pyicloud` already verified that. I currently using a fork-and-customized [icloudpy](https://github.com/leafnsand/icloudpy) and it works well so far. If you are interested in that, you can checkout my public repo. Although it is written with Python but simple enough.
```
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Welcome to icloud-photos-sync, v.1.1.0!
Made with <3 by steilerDev
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Authenticating user...
Device trusted
Sign in successful!
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Experienced fatal error at 7/4/2023, 12:07:45 PM: APP_SYNC (FATAL): Sync failed caused by AUTH_FAILED (FATAL): iCloud Authentication failed caused by ICLOUD_PHOTOS_SETUP_FAILED (FATAL): Unable to get iCloud Photos service ready caused by ICLOUD_PHOTOS_INDEXING_STATE_UNAVAILABLE (FATAL): Unable to get indexing state caused by Request failed with status code 404 (Error Code: b242b6aa-dd14-423d-96f3-4391c8714f1f)
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

[icloud-drive-docker](https://github.com/leafnsand/icloud-drive-docker) is a docker utility which I used for photo syncing, and it support raw+jpeg version syncing already. I wish I can dig deeper into your codebase but I am not very good at TypeScript and JavaScript. But I am glad to share with you my experience of using iCloud in China. Anything could help you, just ask.

**Comment by steilerDev** [comment]
See #286 for iCloud China Support.

**Comment by steilerDev** [comment]
Closing this for now - track iCloud China support in the linked ticket and I will soon start a thread on Library Version 2 which will be a bigger rework that hopefully should give more flexibility on which assets to download.

iCloud China support will be added, based on your comment and a second requestor - afterwards I need you to validate the solution.
